spring-boot-3-integrationtests/src/test/java/org/axonframework/tracing/TestSpanFactory.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.messaging.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Creates spans that record whether they were started, ended or ended in an exception. Useful for tracing logic
+ * assertions in unit tests and provides several out of the box.
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public class TestSpanFactory implements SpanFactory {
```

Comment:
Perhaps we can retrieve this as a test dependency too? If I recall correctly, that's done for the other tests utilizing this `SpanFactory` implementation.

Reply:
Yes, done.